docs/conditions.rst

```diff
@@ -2,37 +2,376 @@
+Once all the conditions have been met, it will return you all the events that matched,
+which can be used in action.
+
+    .. note::
+        Note that this case the engine will consider **all the different events** until meet the conditions,
+        regardless of whether those events come from one or multiple sources.
+        Multiple conditions with ``all`` are not equivalent to a single condition with the ``and`` operator.
+
+        If you want to match only one event using multiple attributes
+        the rule must use a single condition with the ``and`` operator:
+
+    .. code-block:: yaml
+
+        name: One condition combining attributes
+        condition: event.target_os == "linux" and event.tracking_id == 345
+        action:
+          debug:
+
+
+Multiple conditions where **any** one of them has to match
+----------------------------------------------------------
+
+    .. code-block:: yaml
+
+        name: Any condition can match
+        condition:
+          any:
+            - event.target_os == "linux"
+            - event.target_os == "windows"
+        action:
+          debug:
+
+    .. note::
+        Note that this case the engine will consider **all the different events** until meet the conditions,
+        regardless of whether those events come from one or multiple sources.
+        Multiple conditions with ``any`` are not equivalent to a single condition with the ``or`` operator.
+
+        If you want to match only one event using multiple attributes
+        the rule must use a single condition with the ``or`` operator:
+
+    .. code-block:: yaml
+
+        name: One condition combining attributes
+        condition: event.target_os == "linux" or event.target_os == "windows"
+        action:
+          debug:
+
+
+Multiple conditions with facts and events and **all** of one of them have to match
+----------------------------------------------------------------------------------
+
+    .. code-block:: yaml
+
+        name: Condition using both a fact and an event
+        condition:
+          all:
+            - fact.meta.hosts == "localhost"
+            - event.target_os == "windows"
+        action:
+          debug:
+
+Condition with fact and event, fact being passed in via --variables on command line
+-----------------------------------------------------------------------------------
+
+    .. code-block:: yaml
+
+        name: Condition using a passed in variable and an event
+        condition:
+          all:
+            - facts.first << fact.custom.expected_index is defined
+            - event.i == facts.first.custom.expected_index
+        action:
+          debug:
+
+In the above example the custom.expected_index was passed in a variables file via ``--variables`` from the
+command line to ansible-rulebook.
+
+| When evaluating a single event you can compare multiple
+| properties/attributes from the event using **and** or **or**
```

Comment:
```suggestion
| When evaluating a single event you can compare multiple properties/attributes from the event using **and** or **or**.
```

Reply:
suggestion rejected, many things are in the same way, out of the scope of this PR